api/src/main/java/jakarta/data/repository/KeysetPageable.java

```diff
@@ -0,0 +1,328 @@
+ * end of the current page as follows,</p>
+ *
+ * <pre>
+ * page = employees.findByHoursWorkedGreaterThan(1500, page.nextPageable());
+ * </pre>
+ *
+ * <p>Because the page is keyset aware, the <code>KeysetPageable</code>
+ * that it returns from the call to {@link KeysetAwareSlice#nextPageable}
+ * above is based upon a keyset from that page to use as a starting point
+ * after which the results for the next page are to be found.</p>
+ *
+ * <p>You can also construct a <code>KeysetPageable</code> directly, which
+ * allows you to make it relative to a specific list of values. The number and
+ * order of values must match that of the {@link OrderBy} annotations,
+ * <code>Pageable</code> {@link Pageable#of(long, long, Sort...) Sort} parameters,
+ * or <code>OrderBy</code> name pattern of the repository method.
+ * For example,</p>
+ *
+ * <pre>
+ * Employee emp = ...
+ * KeysetPageable pagination = Pageable.size(50).afterKeyset(emp.lastName, emp.firstName, emp.id);
+ * page = employees.findByHoursWorkedGreaterThan(1500, pagination);
+ * </pre>
+ *
+ * <p>By making the query for the next page relative to observed values,
+ * not a numerical position, keyset pagination is less vulnerable to changes
+ * that are made to data in between page requests. Adding or removing entities
+ * is possible without causing unexpected missed or duplicate results.
+ * Keyset pagination does not prevent misses and duplicates if the entity
+ * properties which are the sort criteria for existing entities are modified
+ * or if an entity is re-added with different sort criteria after having
+ * previously been removed.</p>
+ *
+ * <h2>Keyset Pagination with &#64;Query</h2>
+ *
+ * <p>Keyset pagination involves generating and appending to the query
+ * additional query conditions for the keyset properties. In order for that to
+ * be possible, a user-provided
+ * <a href="https://eclipse-ee4j.github.io/jakartaee-tutorial/#full-query-language-syntax">JPQL</a>
+ * query must end with a
+ * <code>WHERE</code> clause to which additional conditions can be appended.
+ * Enclose the entire conditional expression of the <code>WHERE</code> clause
+ * in parenthesis.
+ * Sort criteria must be specified independently from the user-provided query,
+ * either with the {@link OrderBy} annotation or
+ * <code>Pageable</code> {@link Pageable#of(long, long, Sort...) Sort} parameters.
+ * For example,</p>
+ *
+ * <pre>
+ * &#64;Query("SELECT o FROM Customer o WHERE (o.ordersPlaced &gt;= ?1 OR o.totalSpent &gt;= ?2)")
+ * &#64;OrderBy("zipcode")
+ * &#64;OrderBy("birthYear")
+ * &#64;OrderBy("id")
+ * KeysetAwareSlice&lt;Customer&gt; getTopBuyers(int minOrders, float minSpent, Pageable pagination);
+ * </pre>
+ *
+ * <h2>Page Numbers and Totals</h2>
+ *
+ * <p>Page numbers, total numbers of elements across all pages, and total
+ * count of pages are not accurate when keyset pagination is used and should
+ * not be relied upon.</p>
+ */
+public class KeysetPageable extends Pageable {
```

Comment:
As per my question in https://github.com/jakartaee/data/pull/64

Why is this not an interface?

Reply:
> As per my question in #64
> 
> Why is this not an interface?

It's a valid topic to discuss whether `Pageable` (and similarly `KeysetPageable`) ought to be made into an interface, but within this pull it is right for `KeysetPageable` to follow after `Pageable`.  Please open a separate issue to discussing changing to interface for Pageable.

Reply:
This topic is being addressed separately in #65